homeassistant/components/filter/sensor.py

```diff
@@ -400,11 +400,11 @@ class Filter:
 
     def __init__(
         self,
-        name,
-        window_size: int = 1,
-        precision: int | None = None,
-        entity: str | None = None,
-    ):
+        name: str,
+        window_size: int | timedelta,
+        precision: int,
+        entity: str,
```

Comment:
`window_size` can be a `timedelta` (see isinstance check below), and although it could be optional I think it is better to set the default value in the child class instead.
`precision` and `entity` should not be optional - a value is always passed.